src/python/pants/bin/local_pants_runner.py

```diff
@@ -180,15 +183,13 @@ def create(
         # This works as expected due to the encapsulated_logger in DaemonPantsRunner and
         # we don't have to gate logging setup anymore.
 
-        native = Native()
         level = LogLevel.ERROR if getattr(global_options, "quiet", False) else global_options.level
-        setup_logging(
-            level,
-            log_dir=global_options.logdir,
-            native=native,
-            console_stream=sys.stderr,
-            warnings_filter_regexes=global_options.ignore_pants_warnings,
-        )
+        ignores = global_options.ignore_pants_warnings
+        clear_previous_loggers()
+        setup_logging_to_stderr(level, warnings_filter_regexes=ignores)
+        log_dir = global_options.logdir
+        if log_dir:
+            setup_logging_to_file(level, log_dir=log_dir, warnings_filter_regexes=ignores)
```

Comment:
Really nice. I can figure out a high level what is happening without having to look at the function definitions.